packages/gatsby-source-mongodb/src/gatsby-node.js

```diff
@@ -148,6 +156,91 @@ function createNodes(
       })
       resolve()
     })
+  }).then(() => {
+    const changeStream = collection.watch()
+
+    changeStream.on(
+      `change`,
+      ({ operationType, documentKey, fullDocument }) => {
+        switch (operationType) {
+          case `replace`:
+          case `insert`: {
+            const id = idToString(documentKey._id)
+            const nodeId = createNodeId(id)
+
+            // if (preserveObjectIds) {
+            //   for (let key in item) {
+            //     item[key] = stringifyObjectIds(item[key])
+            //   }
+            // }
+
+            const node = {
```

Comment:
We might want to move all of this to a helper since it's a bunch of unnecessary code duplication